src/folder-based-storage-component.ts

```diff
@@ -111,6 +110,14 @@ export async function createFolderBasedFileSystemContentStorage(
       const entries = await Promise.all(cids.map(async (cid): Promise<[string, boolean]> => [cid, await exist(cid)]))
       return new Map(entries)
     },
-    allFileIds: () => allFileIdsRec(root)
+    allFileIds: () => allFileIdsRec(root),
+
+    findKeys: async function* findKeys(prefix?: string): AsyncIterable<string> {
+      for await (const key of allFileIdsRec(root)) {
```

Comment:
This will explode in catalysts with many many storage files

Reply:
Why? Is it because it will take long? Shouldn't being `AsyncIterable` take care of it? Or does that line materialize all the collection in memory? I thought it didn't...

Reply:
I'm gonna remove the new method and leverage the old one (`allFileIds`). See https://github.com/decentraland/catalyst-storage/pull/50. That shouldn't explode.